src/core/algorithms/order/dependency_checker.h

```diff
@@ -0,0 +1,13 @@
+#pragma once
+
+#include <enum.h>
+
+#include "sorted_partitions.h"
+
+namespace algos::order {
+
+BETTER_ENUM(ValidityType, char, valid = 0, merge, swap);
+
+ValidityType CheckForSwap(SortedPartition const& l, SortedPartition const& r);
```

Comment:
Why not a static method of `SortedPartition`?

Reply:
It'll be changed a bit in the parallel version to work not only with `SortedPartition`